### PR TITLE
Update to `ProgressMaterialButton` in `ScheduleAppointmentSheet`

### PR DESCRIPTION
**Pull request by Honey14**
https://app.clubhouse.io/simpledotorg/story/554/add-progress-ui-when-scheduling-an-appointment

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentEffectHandler.kt [hunk 1]
```diff
@@ -25,6 +25,7 @@ import org.simple.clinic.util.toNullable
 import org.simple.clinic.util.toOptional
 import org.simple.clinic.uuid.UuidGenerator
 import java.time.LocalDate
+import java.util.concurrent.TimeUnit
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentEffectHandler.kt:28]: Seems like this is an unused import, can you please remove this?

File: app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentEvent.kt [hunk 2]
```diff
@@ -2,6 +2,7 @@ package org.simple.clinic.scheduleappointment
 
 import org.simple.clinic.facility.Facility
 import org.simple.clinic.overdue.PotentialAppointmentDate
+import org.simple.clinic.widgets.ProgressMaterialButton
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentEvent.kt:5]: Seems like this is an unused import, can you please remove this?

File: app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentSheet.kt [hunk 3]
```diff
@@ -233,4 +235,12 @@ class ScheduleAppointmentSheet : BottomSheetActivity(), ScheduleAppointmentUi, S
   override fun showPatientFacility(facilityName: String) {
     selectedFacilityName.text = facilityName
   }
+
+  override fun showProgressUntilAppointmentScheduled() {
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentSheet.kt:239]: Can we rename this to `showDoneButtonProgress` or `showProgress`?

File: app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentModel.kt [hunk 4]
```diff
@@ -13,7 +14,8 @@ data class ScheduleAppointmentModel(
     val patientUuid: UUID,
     val potentialAppointmentDates: List<PotentialAppointmentDate>,
     val selectedAppointmentDate: PotentialAppointmentDate?,
-    val appointmentFacility: Facility?
+    val appointmentFacility: Facility?,
+    val doneButtonState : ButtonState?
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentModel.kt:18]: We can make it non-null and provide the default state to be `Enabled` in `create`, WDYT?

Comment (Honey14) [reply app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentModel.kt:18]: This makes sense. I'll do it

**Pull request merged**